gen/calc.py

```diff
@@ -608,6 +608,18 @@ def validate_adminrouter_tls_version_present(
     assert enabled_tls_flags_count > 0, msg
 
 
+def validate_adminrouter_x_frame_options(adminrouter_x_frame_options):
+    """
+    Provide a basic validation that checks that provided value starts with
+    one of the supported options: DENY, SAMEORIGIN, ALLOW-FROM
+    See: https://tools.ietf.org/html/rfc7034#section-2.1
+    """
+    msg = 'X-Frame-Options must be set to one of DENY, SAMEORIGIN, ALLOW-FROM'
+    regex = r"^((DENY|SAMEORIGIN)|ALLOW-FROM .+)$"
```

Comment:
The inner parentheses are not required: `|` has the lowest precedence.

According to the RFC the space after `ALLOW-FROM` can also be a tab character.